src/governance.cpp

```diff
@@ -211,13 +211,27 @@ void CGovernanceManager::ProcessMessage(CNode* pfrom, std::string& strCommand, C
 
         if(!fIsValid) {
             if(fMasternodeMissing) {
-                mapMasternodeOrphanObjects.insert(std::make_pair(nHash, object_time_pair_t(govobj, GetAdjustedTime() + GOVERNANCE_ORPHAN_EXPIRATION_TIME)));
+
+                int& count = mapMasternodeOrphanCounter[govobj.GetMasternodeVin().prevout];
+                if (count >= 10) {
+                    LogPrint("gobject", "MNGOVERNANCEOBJECT -- Too many orphan objects, missing masternode=%s\n", govobj.GetMasternodeVin().prevout.ToStringShort());
+                    // ask for this object again in 2 minutes
```

Comment:
We still can receive fake objects from missing masternode with valid collateral and there is no way to verify their signatures unless MNB has received. Because of that we keep no more than 10 such objects for each MN in memory and wait for their MNB's. Then MNB should come and we accept this object otherwise just drop it and apply ban score for node who send it.

If we have already more than 10 orphan objects from certain MN then we don't keep all further objects in memory but ask for them again later.
`CNode::AskFor` already has a mechanism for retrying of asking the object in 2 minutes after the last.